packages/lesswrong/components/notifications/NotificationsPage/NotificationsPageNotification.tsx

```diff
@@ -0,0 +1,176 @@
+import React, { FC } from "react";
+import { Components, registerComponent } from "../../../lib/vulcan-lib";
+import { Link } from "../../../lib/reactRouterWrapper";
+import { postGetPageUrl } from "../../../lib/collections/posts/helpers";
+import { commentGetPageUrlFromIds } from "../../../lib/collections/comments/helpers";
+import { tagGetUrl } from "../../../lib/collections/tags/helpers";
+import { localgroupGetUrl } from "../../../lib/collections/localgroups/helpers";
+import {
+  NotificationDisplay,
+  NotificationType,
+  getNotificationTypeByName,
+} from "../../../lib/notificationTypes";
+import type { ForumIconName } from "../../common/ForumIcon";
+import type { IconVariant } from "./NotificationsPageItem";
+
+const styles = (theme: ThemeType) => ({
+  primaryText: {
+    color: theme.palette.grey[1000],
+  },
+});
+
+type DisplayConfig = {
+  Display: NotificationType["Display"] | null,
+  Icon: ForumIconName | FC,
+  iconVariant: IconVariant,
+}
+
+const getDisplayConfig = ({type, comment}: NotificationDisplay): DisplayConfig => {
+  try {
+    const {Display} = getNotificationTypeByName(type);
+    return {
+      Display,
+      ...(comment
+        ? {Icon: "CommentFilled", iconVariant: "primary"}
+        : {Icon: "DocumentFilled", iconVariant: "grey"}
+      ),
+    };
+  } catch (e) {
+    // eslint-disable-next-line no-console
+    console.error("Invalid notification type:", type, e);
+  }
+
+  return {
+    Display: null,
+    Icon: "DocumentFilled",
+    iconVariant: "grey",
+  };
+}
+
+export const NotificationsPageNotification = ({notification, classes}: {
+  notification: NotificationDisplay,
+  classes: ClassesType<typeof styles>,
+}) => {
+  const {Display, Icon, iconVariant} = getDisplayConfig(notification);
+  if (!Display) {
+    return null;
+  }
+
+  const {
+    createdAt, comment, post, user, tag, localgroup, link, tagRelId,
+  } = notification;
+  const displayUser = (
+    user ??
+    post?.user ??
+    comment?.user
+  ) as UsersMinimumInfo | undefined;
+  const displayPost = post ?? comment?.post;
+  const displayLocalgroup = localgroup ?? post?.group;
+
+  const {UsersName, PostsTooltip, FormatDate, NotificationsPageItem} = Components;
+  const User: FC = () => (
```

Comment:
This is a bit of a weird pattern, but I think it works quite nicely.

We already define templates in the `notificationTypes` file for creating strings describing the notification for the user. Unfortunately, because they're stings, they're not interactive. To make them interactive, we add separate templates in that file that output JSX instead, and they take these placeholders as arguments to insert into the output.